shared/middleware/helpers/firebase_helper.rb

```diff
@@ -26,19 +26,48 @@ def table_as_csv(table_name)
     table_to_csv(records, column_order: ['id'])
   end
 
+  def number?(num)
+    !Float(num).nil? rescue false
+  end
+
+  def csv_as_table(csv_data)
+    records = {}
+    id = 1
+    table = CSV.parse(csv_data, headers: true)
+    table.each do |row|
+      record = OpenStruct.new
+      record.id = id
+      table.headers.each do |col|
+        value = (number? row[col]) ? row[col].to_f : row[col]
+        record[col] = value
+      end
+      records[id] = record.to_h.to_json
+      id += 1
+    end
+    # add id as first column
+    columns = table.headers.unshift('id')
+    return records, columns
+  end
+
   def delete_shared_table(table_name)
-    @firebase.delete("/v3/channels/shared/counters/tables/#{table_name}")
-    @firebase.delete("/v3/channels/shared/storage/tables/#{table_name}/records")
+    response = @firebase.delete("/v3/channels/shared/counters/tables/#{table_name}")
+    return response unless response.success?
```

Comment:
This might be my lack of knowledge on ruby showing through, but wouldn't this `return` prevent the following lines from running?

Reply:
Yes, I think if one of the requests fails, we shouldn't try the next one and just return the error.

Reply:
Ahh ok. This makes more sense. I was reading the `unless` backward